dotnet/dynamodb/README.md

```diff
@@ -0,0 +1,437 @@
+```
+
+Therefore, all of the projects that require a table name use the default table
+**CustomersOrdersProducts** in the default region **us-west-2**.
+Similar values exist for most variable values in all projects.
+This means there are few command-line arguments for any executable.
+
+## General code pattern
+
+It's important that you understand the new async/await programming model in the
+[AWS SDK for .NET](https://docs.aws.amazon.com/amazondynamodb/latest/developerguide).
+
+These code examples use the following NuGet packages:
+
+- AWSSDK.Core, v3.5.0
+- AWSSDK,DynamoDBv2, v3.5.0
+
+## Unit tests
+
+We use [moq4](https://github.com/moq/moq4) to create unit tests with mocked objects.
+All of the code example projects have a companion unit test,
+where the name of the unit test project is the same as the tested project,
+with a **Test** suffix.
+
+You can install the **moq** and **Extensions** unit testing Nuget packages with
+the following commands:
+
+```
+dotnet add package moq
+dotnet add package Microsoft.UnitTestFramework.Extensions
+```
+
+A typical unit test looks something like the following,
+which tests a call to **CreateTableAsync** in the
+**CreateTable** project:
+
+```
+using System;
+using System.Net;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Amazon.DynamoDBv2;
+using Amazon.DynamoDBv2.Model;
+
+using Moq;
+
+using Xunit;
+
+namespace DynamoDBCRUD
+{        
+    public class CreateTableTest
+    {
+        readonly string _tableName = "testtable";
+
+        private IAmazonDynamoDB CreateMockDynamoDBClient()
+        {
+            var mockDynamoDBClient = new Mock<IAmazonDynamoDB>();
+
+            mockDynamoDBClient.Setup(client => client.CreateTableAsync(
+                It.IsAny<CreateTableRequest>(),
+                It.IsAny<CancellationToken>()))
+                .Callback<CreateTableRequest, CancellationToken>((request, token) =>
+                {
+                    if (!string.IsNullOrEmpty(_tableName))
+                    {
+                        bool areEqual = _tableName == request.TableName;
+                        Assert.True(areEqual, "The provided table name is not the one used to create the table");
+                    }
+                })
+                .Returns((CreateTableRequest r, CancellationToken token) =>
+                {
+                    return Task.FromResult(new CreateTableResponse { HttpStatusCode = HttpStatusCode.OK });
+                });
+
+            return mockDynamoDBClient.Object;
+        }
+
+        [Fact]
+        public async Task CheckCreateTable()
+        {
+            IAmazonDynamoDB client = CreateMockDynamoDBClient();
+
+            var result = await CreateTable.MakeTableAsync(client, _tableName);
+
+            bool ok = result.HttpStatusCode == HttpStatusCode.OK;
+            Assert.True(ok, "Could NOT create table " + _tableName);
+        }
+    }
+}
+```
+
+To run this test,
+navigate to the *CreateTableTest* folder and run:
```

Comment:
... the **CreateTableTest** (use bold on folder/directory/UI names)

Reply:
Fixed